src/runtime/wasm.ts

```diff
@@ -667,6 +680,23 @@ export class WasmParticle extends Particle {
     return p;
   }
 
+  // TODO(sjmiles): UiBroker changes ... we don't have `capabilities` yet,
+  // so just go straight to output
+  output(content) {
+    this.container.apiPort.Output(this, content);
+  }
+
+  // render request call-back from wasm
+  onRenderOutput(templatePtr: WasmAddress, modelPtr: WasmAddress) {
+    const content: Content = {templateName: 'default'};
+    content.template = this.container.read(templatePtr);
+    content.model = new StringDecoder().decodeDictionary(this.container.read(modelPtr));
```

Comment:
`StringDecoder.decodeDictionary(...)` should be sufficient.

Reply:
I used the patterns in `renderImpl` and `serviceRequest` ... is it wrong in those places also?

Reply:
No, StringDecoder sets and consumes a member field while decoding.

Reply:
Alex: look at the code. `decodeDictionary()` sets `this.str`, which means it needs an instance, which means `new StringDecoder().decodeDictionary` is correct.